### PR TITLE
gaussian spot detector pipeline component

### PR DESCRIPTION
**Pull request by ambrosejcarr**
This is ready for comment again. 

@dganguli, Jeremy, Nick and I discussed how we'd like to organized spot finding. This branch implements that discussion using a pipeline component. 

I've extracted the unittests into another branch because it's not clear how to host the data for them yet.

### CODE REVIEW

File: starfish/pipeline/features/spot_attributes.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    def __init__(self, spot_attributes):
+        """Construct a SpotAttributes instance
+
+        Parameters
+        ----------
+        spot_attributes : pd.DataFrame
+
+        """
+        missing_fields = {'x', 'y', 'r', 'intensity', 'spot_id'}.difference(spot_attributes.columns)
```
[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply starfish/pipeline/features/spot_attributes.py]: These are the current spec requirements for a SpotAttributes table. I'll made this clearer. Quality goes in the decoder table, I believe. 

Made #190 to address whether the SpotAttribute table needs 3D.

File: starfish/pipeline/features/spots/detector/gaussian.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+from starfish.munge import gather
+from starfish.pipeline.features.encoded_spots import EncodedSpots
+from starfish.pipeline.features.spot_attributes import SpotAttributes
+from ._base import SpotFinderAlgorithmBase
+
+
+class GaussianSpotDetector(SpotFinderAlgorithmBase):
+
+    def __init__(self, min_sigma, max_sigma, num_sigma, threshold, blobs_image_name, measurement_type='max', **kwargs):
```
[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply starfish/pipeline/features/spots/detector/gaussian.py]: It needs some docs!

File: starfish/pipeline/features/spots/detector/gaussian.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    @staticmethod
+    def measure_blob_intensity(image, spots, measurement_function) -> pd.Series:
+        return spots.apply(
+            lambda row: measurement_function(image[int(row.x_min):int(row.x_max), int(row.y_min):int(row.y_max)]),
+            axis=1
+        )
+
+    def encode(self, stack, spot_attributes):
+        # create stack squeeze map
+        stack.squeeze(bit_map_flag=False)  # this is side-effecting :( :(
```
[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply starfish/pipeline/features/spots/detector/gaussian.py]: Good call.

**Comment by ambrosejcarr** [comment]
Airplane wifi did some odd things to my comment responses. Hope they're intelligible.

File: starfish/starfish.py [hunk 4]
```diff
@@ -2,12 +2,12 @@
 
 import argparse
 import cProfile
-import json
 import os
 import sys
 from pstats import Stats
 
-from .pipeline import registration
+from starfish.pipeline.features.spots.detector import SpotFinder
```
Comment (ttung) [inline starfish/starfish.py]: Should make this import style consistent (relative vs absolute).

**Pull request merged**